highway.tex

```diff
@@ -615,9 +630,207 @@ \subsection{Checkpoints}
+
+This problem is made worse by noting that the honest validators must essentially agree on which votes they are going to process. If validator $A$ downloads one of these spammy votes and incorporates it into one of their own justifications, other validators will not be able to process $A$'s vote until they have also processed these other messages. This is perhaps even worse if an attacker sends different sets of spam votes to each of the honest validators. If each honest validator includes a reasonable subset of these messages in their next block, it may not be apparent that the system is being overtaxed until they have a chance to look at each others' blocks and realize that they will have a huge number of new votes that need to be downloaded and processed before they can continue.
+
+In this section, we first outline two of possible spam patterns that creates very high amounts of messages, in order to show that without taking any preventive measures, the spam attacks can cause a resource depletion in validator nodes and hence possibly crash the whole protocol. Then, we show how using relatively simple techniques, it is possible to bound the amount of possible spam to the easily managable size. 
+
+\subsection{Classification of Basic Spam Patterns}
+
+In this subsection, we lay out two basic categories of spam attacks:
+
+In a vertical spam, the attacker does not produce any equivocations and merely produces a single long string of sequential messages. This could be done by simply setting their round exponent to an unrealistically small number. This is substantially the easier of type of spam to deal with, but the only type that can be produced without the attacker equivocating, and thus allowing them to be readily punished for it.
+
+In an equivocation bomb, instead of making many sequential messages, the attacker makes a large number of non-sequential messages. This is a somewhat trickier situation to deal with especially because as discussed above, the attacker can show different messages to different validators, and it will take some coordination between the other validators in order to determine which messages should and shouldn't be ignored.
+
+
+\subsubsection{Vertical Spam}
+
+In vertical spam, an attacker simply creates a long sequence of valid votes, faster than the other validators can readily handle. Perhaps the easiest way to deal with this issue is to set a minimum round exponent that is not too much smaller than what one would expect the fastest rate that the system can process messages. If additionally it is required that messages be timestamped and that no validator sends more than three messages in a round (measured by comparing the timestamp of the message to the timestamp of the third-most-recent message by the same validator in its history), this ensures that no attacker can create valid, non-equivocating messages at a rate much faster than that of any of the honest validators.
```

Comment:
We can even limit it to [two votes per round](https://github.com/CasperLabs/casper-node/blob/b6e2f6c36c8dd0c0155bba0754b02b908eb1dfd6/node/src/components/consensus/highway_core/state.rs#L390): The leader doesn't need to send a confirmation for their own proposal, and the other validators aren't supposed to send a proposal.

Reply:
Indeed, actually two rounds are sufficient also according to the current description in the paper.